service/src/main/java/org/ehrbase/repository/EhrRepository.java

```diff
@@ -217,46 +202,32 @@ public void adminDelete(UUID ehrId) {
     @Transactional
     public void update(UUID ehrId, EhrStatus ehrStatus, @Nullable UUID contributionId, @Nullable UUID auditId) {
 
-        Result<EhrStatusVersionHistoryRecord> headRecords =
-                findVersionHeadRecords(singleEhrStatusCondition(ehrId, tables.versionHead()));
-
-        EhrStatusVersionHistoryRecord versionHead = headRecords.getFirst();
-        if (versionHead.getSysVersion() + 1 != extractVersion(ehrStatus.getUid())) {
-            throw new PreconditionFailedException(NOT_MATCH_LATEST_VERSION);
-        }
-
-        copyHeadToHistory(versionHead, timeProvider.getNow());
-        deleteHead(
-                singleEhrStatusCondition(ehrId, tables.versionHead()),
-                versionHead.getSysVersion(),
-                PreconditionFailedException::new);
-
-        commitHead(
+        update(
                 ehrId,
                 ehrStatus,
+                singleEhrStatusCondition(ehrId, tables.versionHead()),
+                singleEhrStatusCondition(ehrId, tables.versionHistory()),
                 contributionId,
                 auditId,
-                ContributionChangeType.modification,
                 r -> {},
-                r -> r.setEhrId(ehrId));
+                r -> r.setEhrId(ehrId),
+                "No EHR_STATUS in ehr: %s".formatted(ehrId));
     }
 
     public Optional<VersionedEhrStatus> getVersionedEhrStatus(UUID ehrId) {
 
-        return findRootRecordByVersion(ehrId, 1).map(root -> {
-            VersionedEhrStatus versionedComposition = new VersionedEhrStatus();
-            versionedComposition.setUid(new HierObjectId(root.getVoId().toString()));
-            versionedComposition.setOwnerId(new ObjectRef<>(new HierObjectId(ehrId.toString()), "local", "ehr"));
-            versionedComposition.setTimeCreated(new DvDateTime(root.getSysPeriodLower()));
-            return versionedComposition;
-        });
-    }
-
-    private Optional<EhrStatusVersionHistoryRecord> findRootRecordByVersion(UUID ehrId, int version) {
         return findRootRecordByVersion(
-                singleEhrStatusCondition(ehrId, tables.versionHead()),
-                singleEhrStatusCondition(ehrId, tables.versionHistory()),
-                version);
+                        singleEhrStatusCondition(ehrId, tables.versionHead()),
+                        singleEhrStatusCondition(ehrId, tables.versionHistory()),
+                        1)
+                .map(root -> {
+                    VersionedEhrStatus versionedComposition = new VersionedEhrStatus();
+                    versionedComposition.setUid(new HierObjectId(root.getVoId().toString()));
+                    versionedComposition.setOwnerId(
+                            new ObjectRef<>(new HierObjectId(ehrId.toString()), "local", "ehr"));
+                    versionedComposition.setTimeCreated(new DvDateTime(root.getSysPeriodLower()));
+                    return versionedComposition;
```

Comment:
This would be a nice private static method

Reply:
moved out to static method